src/quicklist.c

```diff
@@ -313,6 +318,14 @@ void quicklistRepr(unsigned char *ql, int full) {
  * If compress depth is larger than the entire list, we return immediately. */
 REDIS_STATIC void __quicklistCompress(const quicklist *quicklist,
                                       quicklistNode *node) {
+    quicklistNode *forward = quicklist->head;
+    quicklistNode *reverse = quicklist->tail;
+
+    if (forward) forward->recompress = 0;
+    if (reverse) reverse->recompress = 0;
+    quicklistDecompressNode(forward);
+    quicklistDecompressNode(reverse);
```

Comment:
I'm not sure why it's necessary to uncompress head and tail every time, it might hide what's really going wrong.

Reply:
before my change you had : 
    while (depth++ < quicklist->compress) {
        quicklistDecompressNode(forward);
        quicklistDecompressNode(reverse);

so after pop new head / tail will be decompressed in case  quicklist->compress > 0 
what happens if we change  quicklist->compress ?